### PR TITLE
refactor(xy): adjust some configurations to optimize chart display

### PR DESCRIPTION
**Pull request by eraop**
## Description

> Adjust some configurations to optimize chart display.

## Screenshots

- [x] This PR does not have any UI modifications  

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation or example update

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation and example

### CODE REVIEW

File: src/lib/xy/xy.ts [hunk 1]
```diff
@@ -269,7 +280,7 @@ export abstract class XYChart extends Chart<DataTableLike, XYChartOptions> {
         dataset.tension = styleMapping.tension;
       }
     }
-    return this.afterDatasetCreated(dataset, { styleOptions: styleMapping }, colors[index], index);
+    return this.afterDatasetCreated(dataset, { styleOptions: styleMapping, color: colors[index], index: index });
```
Comment (bndynet) [inline src/lib/xy/xy.ts:283]: You can ignore the first `index:` if it is duplicated.

File: src/helpers/utils.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    case ChartType.Gauge:
+    case ChartType.Donut:
+      chartType = 'doughnut';
+      break;
+    default:
+      chartType = 'bar';
+      break;
+  }
+  return chartType;
+}
```
Comment (justinZan) [inline src/helpers/utils.ts:28]: Good Job

File: src/lib/legend/legend.ts [hunk 3]
```diff
@@ -19,7 +20,61 @@ const legendClickHandler = function (
...
 
-export default legendClickHandler;
+const initLegendOptions = function <TOptions extends ChartOptions>(
+  type: ChartType,
+  options: TOptions,
+  selectedLegends?: LegendItemOptions[],
+): _DeepPartialObject<LegendOptions<ChartType>> {
+  return {
+    display: options.legend?.display ?? true,
+    position: options.legend?.position ?? 'top',
```
Comment (justinZan) [inline src/lib/legend/legend.ts:33]: `top` is hardcode

File: src/lib/legend/legend.ts [hunk 4]
```diff
@@ -19,7 +20,61 @@ const legendClickHandler = function (
...
+    labels: {
+      pointStyle: 'rectRounded',
+      usePointStyle: true,
+      generateLabels(chart: Chart<ChartType>) {
+        let labels;
+        if (type === 'pie' || type === 'doughnut') {
```
Comment (justinZan) [inline src/lib/legend/legend.ts:39]: `ChartType.Pie` || `ChartType.doughnut`

**Review by bndynet — APPROVED** [review]

**Pull request merged**